src/blackmetal/src/communication/client/RobotDataReceiver.cpp

```diff
@@ -141,15 +143,21 @@ void RobotDataReceiver::workerThread()
 	};
 
 	while (m_connected) {
-		message = m_queue->pop();
+		robotRequest = m_queue->pop();
 
-		if (message.find("Command\":6") != std::string::npos) {
+		if (robotRequest.command() == bm::Command::GET_LR_WHEEL_VELOCITY) {
 			getSpeedCommand = true;
 		}
-		else if (message.find("Command") != std::string::npos) {
+		else if (robotRequest.command() == bm::Command::SET_LR_WHEEL_VELOCITY) {
+			if (robotRequest.rightWheel() != lastWheelSpeeds.first || robotRequest.leftWheel() != lastWheelSpeeds.second) {
```

Comment:
ak porovnavas float, tak radsej:
```suggestion
			if (fabs(robotRequest.rightWheel() - lastWheelSpeeds.first) < 0.001 || fabs(robotRequest.leftWheel() - lastWheelSpeeds.second) < 0.001) {
```
Mimochodom lastWheelSpeeds je std::pair, mozno by viac citatelna bola custom struktura s atributami left a right. A zaroven robotRequest by mohol vraciat rovnaky datovy typ ako je lastWheelSpeeds a mozes si overidnut operator na porovnavanie. To len taky hint